### ISSUE TITLE
Query types decapitalised

### ISSUE THREAD

**Issue by magaton** [ExpediaGroup/graphql-kotlin#1122]
Hello,
I have the following schema:

```
type Month implements Period {
   id: String!
   value: Int!
}
```

but the augmented (generated schema) decapitalizes the type, so instead of `Month`, I get `month` as the generated query.
Is there any config where I can change this behaviour?

**Comment by smyrick** [comment]
Hello @magaton,

I am assuming you are trying to use `graphql-kotlin-client` and then use one of the build plugins to generate your client code from a query file.

What you have shared it just one type from your schema file. Can you share the query file you are using as well as what plugin and the config you are using for it (Maven or Gradle)?

**Comment by magaton** [comment]
Hi @smyrick 
No, I am playing with spring-boot app that is internally using `neo4j-graphql-java` and `graphql-kotlin-server`.
https://github.com/neo4j-graphql/neo4j-graphql-java/blob/master/examples/graphql-spring-boot/src/main/kotlin/org/neo4j/graphql/examples/graphqlspringboot/config/GraphQLConfiguration.kt#L53

Your SchemaGeneratorConfig has been passed in bean constructor, so I am assuming schema gets built / augmented in `graphql-kotlin-server`.

Example of spring boot app with maven build is at https://github.com/neo4j-graphql/neo4j-graphql-java/tree/master/examples/graphql-spring-boot.

In a nutshell, Types are decapitalized when query types are generated, in your lib, I presume.

**Comment by smyrick** [comment]
Can you share the Kotlin code that you are using to generate the schema then? We do not change the capitalization on the schema

**Comment by magaton** [comment]
Hi @smyrick 
I posted above:
the app is https://github.com/neo4j-graphql/neo4j-graphql-java/tree/master/examples/graphql-spring-boot
It has 2 kotlin config classes and the main class, nothing else.

**Comment by smyrick** [comment]
That application doesn't have any Kotlin code with a `Month` type

**Comment by smyrick** [comment]
We don't modify the case of type in the schema. If you have a Kotlin class like so

```kotlin
class Query {
    fun month(): Month = getMonth()

    data class Month(val id: String, val value: Int)
}
```

this will be the generated schema

```graphql
type Query {
    month: Month!
}

type Month {
    id: String!
    value: Int!
}
```

And you would query it like this

```graphql
{
    month {
        id
    }
}
```

If you can share a project with just `graphql-kotlin-spring-server` that reproduces this capitalization issue I would be happy to fix any bug as quickly as possible, but it looks like your issue may be in some other dependency

**Comment by magaton** [comment]
Indeed.
I replaced the sample schema.graphql with my types (hence Month).
Anyway please start the app as it is and check the generated schema: team, membership and player query types. They are decapitalized.

This is what I see:
```
type Query {
  membership(
    prop: String
    filter: _MembershipFilter
    first: Int
    offset: Int
    orderBy: [_MembershipOrdering!]
  ): [Membership!]!
  player(
    id: ID
    name: String
    filter: _PlayerFilter
    first: Int
    offset: Int
    orderBy: [_PlayerOrdering!]
  ): [Player!]!
  team(
    id: ID
    name: String
    filter: _TeamFilter
    first: Int
    offset: Int
    orderBy: [_TeamOrdering!]
  ): [Team!]!
  echo(string: String!): String!
  pojo(param: String!): Pojo!
}

```

**Comment by smyrick** [comment]
Those are not de capitalized, that is the name of the top level field that you query, and it has the return type

```
pojo(param: String!): Pojo!
```

`pojo` is the field name, `Pojo` is the type, and `!` means it is non-null

**Comment by magaton** [comment]
The top-level fields should be (at least I expect so since this is how it is working in neo4j-graphql-js)
Team, Player, Membership, Pojo, why they are decapitalized?

**Comment by smyrick** [comment]
You are not defining a `Query` object so it must be auto-generated. 

I assuming then that this line here: https://github.com/neo4j-graphql/neo4j-graphql-java/blob/bc88551abe5c5833e268410ed52141a404a44e93/examples/graphql-spring-boot/src/main/kotlin/org/neo4j/graphql/examples/graphqlspringboot/config/GraphQLConfiguration.kt#L80

`neo4jSchema.queryType` returns a type that has a lower case field matches to your type name. That is not our library, but having lowercase top-level fields is using the most used pattern instead of uppercase

**Comment by smyrick** [comment]
So the documentation may also need to be updated on neo4j-graphql too

**Comment by magaton** [comment]
I cannot verify what you are saying.
If I disable both springSchema and mergedSchema that you are pointing to and leave only neo4jSchema @Bean, I am getting lowercased query type fields.
```
type Query {
  membership(
    prop: String
    filter: _MembershipFilter
    first: Int
    offset: Int
    orderBy: [_MembershipOrdering!]
  ): [Membership!]!
  player(
    id: ID
    name: String
    filter: _PlayerFilter
    first: Int
    offset: Int
    orderBy: [_PlayerOrdering!]
  ): [Player!]!
  team(
    id: ID
    name: String
    filter: _TeamFilter
    first: Int
    offset: Int
    orderBy: [_TeamOrdering!]
  ): [Team!]!
}
```

**Comment by smyrick** [comment]
> I am getting lowercased query type fields.

No, those are the query fields (`membership`, `player`, `team`), not the type names (`Membership`, `Player`, `Team`).

This is not an issue with our library, it is with some setup of neo4j or the graphql wrapper you are using. If you can figure out how the `Query` type is being generated, and the field names they use, that is where you will need to make a feature suggestion.

If you were using just `graphql-kotlin-spring-server` you could write this file and we don't change anything

```kotlin
class Query {
  fun lowercase() = "hello"
  fun UPPERCASE() = "hello"
  fun mIxEDcaSE() = "hello"
}
```

**Comment by magaton** [comment]
Yes, you were right. It was a problem with neo4j-graphql-java.
Apologies for the noise.
